From 28fcc000a9643f5b55efbac440e77b9f847bea3f Mon Sep 17 00:00:00 2001
Message-Id: <28fcc000a9643f5b55efbac440e77b9f847bea3f.1615936066.git.marcel.ziswiler@toradex.com>
From: Marcel Ziswiler <marcel.ziswiler@toradex.com>
Date: Tue, 16 Mar 2021 22:03:49 +0100
Subject: [PATCH] arm64: dts: imx8mm-verdin: integrate dsi to hdmi adapter v1.1
 support

- explicitly disable backlight in development board device tree to
  avoid conflict in GPIO3.IO[3] usage
- DSI to HDMI bridge Lontium LT8912 reset was on SODIMM 208 aka Verdin
  GPIO2 which is i.MX 8M Mini SoC ball SPDIF_EXT_CLK aka active-low
  gpio5.IO[5], now connected on SODIMM 21 aka Verdin GPIO_10_DSI which
  is i.MX 8M Mini SoC ball NAND_CE2_B aka active-low gpio3.IO[3]
- re-add GPIO1/GPIO2 to iomuxc default pinctrl
- enable pull-up on GPIO5.IO[5] to at least un-reset former v1.0 DSI to
  HDMI adapters even though reset timing may likely be violated

Related-to: TEI-708
Signed-off-by: Marcel Ziswiler <marcel.ziswiler@toradex.com>
---
 .../boot/dts/freescale/fsl-imx8mm-verdin-dev.dtsi    |  2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin.dtsi | 12 +++++++-----
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin-dev.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin-dev.dtsi
index 66a8d6c49a19..b8ab5242beb3 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin-dev.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin-dev.dtsi
@@ -7,7 +7,7 @@
 	brightness-levels = <0 45 63 88 119 158 203 255>;
 	default-brightness-level = <4>;
 	pwms = <&pwm1 0 6666667 PWM_POLARITY_INVERTED>;
-	status = "okay";
+	status = "disabled";
 };
 
 &csi1_bridge {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin.dtsi
index d0027243d9a6..40a0052eb0dd 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-verdin.dtsi
@@ -645,10 +645,10 @@
 		ddc-i2c-bus = <&i2c2>;
 		hpd-gpios = <&gpio3 15 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_gpio_hpd>, <&pinctrl_gpio1>,
-			    <&pinctrl_gpio2>;
+		pinctrl-0 = <&pinctrl_dsi_bkl_en>, <&pinctrl_gpio_hpd>;
 		reg = <0x48>;
-		reset-gpios = <&gpio5 5 GPIO_ACTIVE_LOW>;
+		/* Verdin GPIO_10_DSI (SODIMM 21) */
+		reset-gpios = <&gpio3 3 GPIO_ACTIVE_LOW>;
 		status = "disabled";
 
 		port {
@@ -873,7 +873,8 @@
 
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_gpio3>, <&pinctrl_gpio4>, <&pinctrl_gpio7>,
+	pinctrl-0 = <&pinctrl_gpio1>, <&pinctrl_gpio2>,
+		    <&pinctrl_gpio3>, <&pinctrl_gpio4>, <&pinctrl_gpio7>,
 		    <&pinctrl_gpio8>, <&pinctrl_gpio_hog1>,
 		    <&pinctrl_gpio_hog2>, <&pinctrl_gpio_hog3>,
 		    <&pinctrl_sai5>;
@@ -902,6 +903,7 @@
 		>;
 	};
 
+	/* Verdin GPIO_10_DSI */
 	pinctrl_dsi_bkl_en: dsi_bkl_en {
 		fsl,pins = <
 			MX8MM_IOMUXC_NAND_CE2_B_GPIO3_IO3	0x1c4		/* SODIMM 21 */
@@ -989,7 +991,7 @@
 
 	pinctrl_gpio2: gpio2grp {
 		fsl,pins = <
-			MX8MM_IOMUXC_SPDIF_EXT_CLK_GPIO5_IO5	0x184		/* SODIMM 208 */
+			MX8MM_IOMUXC_SPDIF_EXT_CLK_GPIO5_IO5	0x1c4		/* SODIMM 208 */
 		>;
 	};
 
-- 
2.26.2

